marklogic-client-api/src/main/java/com/marklogic/client/impl/SSLUtil.java

```diff
@@ -40,29 +51,140 @@ static X509TrustManager getDefaultTrustManager() {
 	 * @return a non-empty array of TrustManager instances based on the JVM's default trust manager algorithm, with the
 	 * first trust manager guaranteed to be an instance of X509TrustManager.
 	 */
-	static TrustManager[] getDefaultTrustManagers() {
+	public static TrustManager[] getDefaultTrustManagers() {
 		final String defaultAlgorithm = TrustManagerFactory.getDefaultAlgorithm();
+		return getTrustManagers(defaultAlgorithm, null);
+	}
+
+	/**
+	 * @param trustManagerAlgorithm e.g. "SunX509".
+	 * @param optionalKeyStore      if not null, used to initialize the TrustManagerFactory constructed based on the
+	 *                              given algorithm.
+	 * @return
+	 */
+	public static TrustManager[] getTrustManagers(String trustManagerAlgorithm, KeyStore optionalKeyStore) {
 		TrustManagerFactory trustManagerFactory;
 		try {
-			trustManagerFactory = TrustManagerFactory.getInstance(defaultAlgorithm);
+			trustManagerFactory = TrustManagerFactory.getInstance(trustManagerAlgorithm);
 		} catch (NoSuchAlgorithmException e) {
-			throw new RuntimeException("Unable to obtain trust manager factory using JVM's default trust manager algorithm: " + defaultAlgorithm, e);
+			throw new RuntimeException(
+				"Unable to obtain trust manager factory using algorithm: " + trustManagerAlgorithm, e);
 		}
 
 		try {
-			trustManagerFactory.init((KeyStore) null);
+			trustManagerFactory.init(optionalKeyStore);
 		} catch (KeyStoreException e) {
-			throw new RuntimeException("Unable to initialize trust manager factory obtained using JVM's default trust manager algorithm: " + defaultAlgorithm
-				+ "; cause: " + e.getMessage(), e);
+			throw new RuntimeException(String.format(
+				"Unable to initialize trust manager factory obtained using algorithm: %s; cause: %s",
+				trustManagerAlgorithm, e.getMessage()), e);
 		}
 
 		TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
 		if (trustManagers == null || trustManagers.length == 0) {
-			throw new RuntimeException("No trust managers found using the JVM's default trust manager algorithm: " + defaultAlgorithm);
+			throw new RuntimeException("No trust managers found using algorithm: " + trustManagerAlgorithm);
 		}
 		if (!(trustManagers[0] instanceof X509TrustManager)) {
 			throw new RuntimeException("Default trust manager is not an X509TrustManager: " + trustManagers[0]);
 		}
 		return trustManagers;
 	}
+
+	/**
+	 * Captures the oft-repeated boilerplate Java code for creating an SSLContext based on a key store.
```

Comment:
This is the location of the main annoying code that a user would otherwise have to write themselves.